osf/db/router.py

```diff
@@ -24,19 +28,28 @@ def _get_master(self):
         Finds the first database that's writeable and returns the configuration name.
         :return: :str: name of database config or None
         """
-        for name, dsn in self.DSNS.iteritems():
-            conn = self._get_conn(dsn)
-            cur = conn.cursor()
-            cur.execute('SHOW transaction_read_only;')  # 'on' for slaves, 'off' for masters
-            row = cur.fetchone()
-            if row[0] == u'off':
+        conn = False
+        while not conn:
+            for name, dsn in self.DSNS.iteritems():
+                try:
+                    conn = self._get_conn(dsn)
+                except Exception as ex:
+                    conn = False
+                    continue
```

Comment:
This skips the rest of the loop, including the `time.sleep` call, which was presumably included to prevent this from executing hundreds of times per second. The settings check, log, and `sleep` call should be happening above this.